packages/core/src/telemetry/sdk.ts

```diff
@@ -80,11 +135,45 @@ function parseOtlpEndpoint(
   }
 }
 
-export function initializeTelemetry(config: Config): void {
+export async function initializeTelemetry(
+  config: Config,
+  credentials?: JWTInput,
+): Promise<void> {
   if (telemetryInitialized || !config.getTelemetryEnabled()) {
     return;
   }
 
+  if (config.getTelemetryUseCollector() && config.getTelemetryUseCliAuth()) {
+    debugLogger.error(
+      'Telemetry configuration error: "useCollector" and "useCliAuth" cannot both be true. ' +
+        'CLI authentication is only supported with in-process exporters. ' +
+        'Disabling telemetry.',
+    );
+    return;
+  }
+
+  // If using CLI auth and no credentials provided, defer initialization
+  if (config.getTelemetryUseCliAuth() && !credentials) {
+    // Register a callback to initialize telemetry when the user logs in.
+    // This is done only once.
+    if (!callbackRegistered) {
+      callbackRegistered = true;
+      authEvents.on('post_auth', async (newCredentials: JWTInput) => {
```

Comment:
shutdownTelemetry needs to remove the listener otherwise this could cause undefined behavior.